lib/Doctrine/DBAL/Driver/SQLAnywhere/SQLAnywhereException.php

```diff
@@ -28,14 +28,6 @@ class SQLAnywhereException extends AbstractDriverException
      */
     public static function fromSQLAnywhereError($conn = null, $stmt = null)
     {
-        if ($conn !== null && ! is_resource($conn)) {
```

Comment:
Judging by the number of newly reported issues, we enforce union types only in a couple of places. Let's revert them for consistency and rely on static analysis rather than on runtime checks.